sky/jobs/scheduler.py

```diff
@@ -184,7 +158,8 @@ def maybe_schedule_next_jobs(pool: Optional[str] = None) -> None:
                 # an ALIVE_WAITING job, but we would be able to launch a WAITING
                 # job.
```

Comment:
This comment is no longer true, right? There could be a WAITING job using a pool that can start (doesn't need a LAUNCHING slot), but the ALIVE_WAITING job we fetch is not on a pool and cannot go to LAUNCHING.